toucan_connectors/snowflake/snowflake_connector.py

```diff
@@ -267,32 +271,41 @@ def _execute_query(self, cursor, query: str, query_parameters: Dict):
         # Prevent error with dict and array values in the parameter object
         query = convert_to_printf_templating_style(query)
         converted_query, ordered_values = convert_to_qmark_paramstyle(query, query_parameters)
+
+        # execute query with timeout.
+        # cf https://docs.snowflake.com/en/user-guide/python-connector-example.html
         query_res = cursor.execute(converted_query, ordered_values)
+        if 'SELECT' in query.upper():
```

Comment:
Why do we have 2 cases regarding the presence or not of SELECT?

Reply:
Do we have examples of queries that will not use select? I think this would be the case for writes (insert, drop, etc.) which we don't want to support using connectors for now.
So my guess is that we could drop the else case

Reply:
on peut avoir des query `show xxx`